Background image in SCSS not translated correctly for production builds

<!--
IF YOU DON'T FILL OUT THE FOLLOWING INFORMATION YOUR ISSUE MIGHT BE CLOSED WITHOUT INVESTIGATING
-->
### Bug Report or Feature Request (mark with an `x`)
```
- [x] bug report -> please search issues before submitting
- [ ] feature request
```

### Versions.
```
    _                      _                 ____ _     ___
   / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
  / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
 / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
/_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
               |___/
@angular/cli: 1.0.0
node: 7.9.0
os: darwin x64
@angular/animations: 4.0.2
@angular/common: 4.0.2
@angular/compiler: 4.0.2
@angular/core: 4.0.2
@angular/forms: 4.0.2
@angular/http: 4.0.2
@angular/material: 2.0.0-beta.2
@angular/platform-browser: 4.0.2
@angular/platform-browser-dynamic: 4.0.2
@angular/platform-server: 4.0.2
@angular/router: 4.0.2
@angular/cli: 1.0.0
@angular/compiler-cli: 4.0.2
```
macOS (Sierra v10.12.4)


### Repro steps.
I have a working site that was running on angular 2.4.10 and angular-cli 1.0.0-beta.26. I read all the change logs and upgrade guides today and managed to upgrade the site to angular 4.0.2 and angular-cli 1.0.0. However, I am experiencing a couple of issues, one of which is reported here.

My landing page has a background image that is specified within a components SCSS file as:
```
#banner {
  background: url("/assets/img/background.jpg") no-repeat center;
  background-size: cover;
  height: 700px;
  padding: 0;
  color: white;
}
```

When I run `ng build --prod --aot` and transfer the dist files to my production server then my site loads but this background image seems to be missing. On inspecting the page I noticed that the CLI has translated the **background** element to:
```
background: url(/assets/img/background.jpg) no-repeat 50%;
```
Notice that the double quotes around the **url** function parameter are missing and that **center** has been translated to **50%**.

If I run `ng serve` instead to test my site locally then I see this element getting translated correctly. However, in this case, none of my assets get loaded as the local build seems to have an issue with how it creates the assets in memory (I think). I added some comments to a related issue to describe this other failure here: https://github.com/angular/angular-cli/issues/5268

### The log given by the failure.
None

### Desired functionality.
SCSS **background** element should be translated to:
```
  background: url("/assets/img/background.jpg") no-repeat center;
```
as it did with angular-cli **v1.0.0-beta.26**.


### Mention any other details that might be useful.

`npm ls cssnano` gives:
```
go2travel-ng-cli@0.0.0 /Users/anjum/Documents/workspace/go2travel-ng-cli
├─┬ @angular/cli@1.0.0
│ └── cssnano@3.10.0 
├── UNMET PEER DEPENDENCY @angular/common@4.0.2
├── UNMET PEER DEPENDENCY @angular/core@4.0.2
├── UNMET PEER DEPENDENCY @angular/http@4.0.2
└── UNMET PEER DEPENDENCY tslint@5.1.0

npm ERR! peer dep missing: @angular/common@^2.3.0, required by md2@0.0.17-2
npm ERR! peer dep missing: @angular/core@^2.3.0, required by md2@0.0.17-2
npm ERR! peer dep missing: @angular/http@^2.3.0, required by md2@0.0.17-2
npm ERR! peer dep missing: tslint@^4.0.0, required by codelyzer@2.1.1
```

`npm ls postcss-normalize-url` gives:
```
go2travel-ng-cli@0.0.0 /Users/anjum/Documents/workspace/go2travel-ng-cli
├─┬ @angular/cli@1.0.0
│ └─┬ cssnano@3.10.0
│   └── postcss-normalize-url@3.0.8 
├── UNMET PEER DEPENDENCY @angular/common@4.0.2
├── UNMET PEER DEPENDENCY @angular/core@4.0.2
├── UNMET PEER DEPENDENCY @angular/http@4.0.2
└── UNMET PEER DEPENDENCY tslint@5.1.0

npm ERR! peer dep missing: @angular/common@^2.3.0, required by md2@0.0.17-2
npm ERR! peer dep missing: @angular/core@^2.3.0, required by md2@0.0.17-2
npm ERR! peer dep missing: @angular/http@^2.3.0, required by md2@0.0.17-2
npm ERR! peer dep missing: tslint@^4.0.0, required by codelyzer@2.1.1
```

--- a/packages/@angular/cli/models/webpack-configs/styles.ts
+++ b/packages/@angular/cli/models/webpack-configs/styles.ts
@@ -47,8 +47,16 @@ export function getStylesConfig(wco: WebpackConfigOptions) {
   const deployUrl = wco.buildOptions.deployUrl || '';
 
   const postcssPluginCreator = function() {
+    // safe settings based on: https://github.com/ben-eb/cssnano/issues/358#issuecomment-283696193
+    const importantCommentRe = /@preserve|@license|[@#]\s*source(?:Mapping)?URL|^!/i;
+    const minimizeOptions = {
+      autoprefixer: false, // full pass with autoprefixer is run separately
+      safe: true,
+      mergeLonghand: false, // version 3+ should be safe; cssnano currently uses 2.x
+      discardComments : { remove: (comment: string) => !importantCommentRe.test(comment) }
+    };
+
     return [
-      autoprefixer(),
       postcssUrl({
         url: (URL: string) => {
           // Only convert root relative URLs, which CSS-Loader won't process into require().
@@ -69,9 +77,10 @@ export function getStylesConfig(wco: WebpackConfigOptions) {
             return `/${baseHref}/${deployUrl}/${URL}`.replace(/\/\/+/g, '/');
           }
         }
-      })
+      }),
+      autoprefixer(),
     ].concat(
-        minimizeCss ? [cssnano({ safe: true, autoprefixer: false })] : []
+        minimizeCss ? [cssnano(minimizeOptions)] : []
     );
   };
   (postcssPluginCreator as any)[postcssArgs] = {

